### PR TITLE
JAVA-8427 migrate maven plugin to sdk repo

### PR DESCRIPTION
**Pull request by BrianPhillips2020**
Migrate maven plugin to sdk repo.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Project
+*.private.env.json
```
Comment (BrianPhillips2020) [inline .gitignore:2]: from the maven plugin's .gitignore

File: README.md [hunk 2]
```diff
@@ -1,80 +1,13 @@
-# Contrast Java SDK
+# Contrast SDK Repo
```
Comment (BrianPhillips2020) [inline README.md:1]: This README just changed, moved the SDK README to the sdk project

File: gradle-plugin/.gitignore
Comment (stiemannkj1) [inline gradle-plugin/.gitignore:1]: Nit: recommend just combining all the `.gitignore` and only having 1 at the root of the project.

**Review by stiemannkj1 — APPROVED** [review]

File: gradle-plugin/.gitignore
Comment (BrianPhillips2020) [reply gradle-plugin/.gitignore:1]: I tried this and they weren't applying properly, I can try again

Comment (BrianPhillips2020) [reply gradle-plugin/.gitignore:1]: I tried to do this and they weren't applying properly

File: .github/workflows/build-maven-plugin.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        with:
+          java-version: 17
+          distribution: temurin
+
+      - uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: temurin
+
+      - name: Maven Verify
```
Comment (BrianPhillips2020) [inline .github/workflows/build-maven-plugin.yml:31]: The maven plugin has an e2e test with TeamServer where it uses credentials stored as secrets. The old credentials were Gilday's and are no longer usable. I think we should keep these tests but we do need to add new credentials to the repo in order to run them.

**Review by rszewczyk — APPROVED** [review]

File: .github/workflows/build-maven-plugin.yml (hunk 3 above)
Comment (BrianPhillips2020) [reply .github/workflows/build-maven-plugin.yml:31]: we made a new test user and put the credentials in the secrets

File: gradle-plugin/.gitignore
Comment (BrianPhillips2020) [reply gradle-plugin/.gitignore:1]: I fixed it, I just had to alter the expected paths for the `.idea` files being ignored

**Pull request merged**